src/components/Collapse.ts

```diff
@@ -1,29 +1,49 @@
 import { useUniqueId } from '@terra-ui-packages/components';
 import _ from 'lodash/fp';
-import { useEffect, useRef, useState } from 'react';
-import { div, h } from 'react-hyperscript-helpers';
-import { Link } from 'src/components/common';
+import { CSSProperties, PropsWithChildren, ReactNode, useEffect, useRef, useState } from 'react';
+import { div, h, HTMLElementProps, WithDataAttributes } from 'react-hyperscript-helpers';
+import { Link, LinkProps } from 'src/components/common';
 import { icon } from 'src/components/icons';
 import colors from 'src/libs/colors';
 import * as Style from 'src/libs/style';
 
-const Collapse = ({
-  title,
-  hover,
-  tooltip,
-  tooltipDelay,
-  summaryStyle,
-  detailsStyle,
-  initialOpenState,
-  children,
-  titleFirst,
-  afterTitle,
-  onFirstOpen = () => {},
-  noTitleWrap,
-  disabled = false,
-  ...props
-}) => {
-  const [isOpened, setIsOpened] = useState(initialOpenState);
+type DivProps = Omit<WithDataAttributes<HTMLElementProps<'div'>>, 'children'>;
+
+interface CollapseProps {
```

Comment:
Since props are passed through to the underlying `div`, suggest

```suggestion
type DivProps = JSX.InstrinsicElements['div'];

interface CollapseProps extends Omit<DivProps, 'title'> {
```

Then `style`, `onClick`, and 'children' don't need to specified again.